api/server/sdk/server.go

```diff
@@ -477,6 +479,13 @@ func (s *sdkGrpcServer) Start() error {
 		s.log.Info("SDK TLS disabled")
 	}
 
+	// Clients will always keep connections open with KeepAlive,
+	// however we will always recycle these after 15mins.
+	// This will handle CSI Provisioner leader changes and force regular new connections.
+	opts = append(opts, grpc.KeepaliveParams(keepalive.ServerParameters{
+		MaxConnectionAge: 15 * time.Minute,
```

Comment:
Ok, I'll remove the keepalive and add this.

Reply:
Thanks! I am just wary that this could lead to a goroutine leak. Long time back we had a goroutine leak in SDK when we were not closing the client connection.

Reply:
@adityadani I've added custom connection cleanup logic. Ready for re-review.

Test results (with modified cleanup intervals)
```
@ggriffiths-k8s-1-node0 portworx[14553]: time="2022-06-08T23:34:07Z" level=info msg="Cleaning up open gRPC connections for CSI distributed provisioning" component=csi-driver
@ggriffiths-k8s-1-node0 portworx[14553]: time="2022-06-08T23:34:07Z" level=info msg="Cleaned up 0 connections for CSI distributed provisioning. 4 connections remaining" component=csi-driver
@ggriffiths-k8s-1-node0 portworx[14553]: time="2022-06-08T23:39:07Z" level=info msg="Cleaning up open gRPC connections for CSI distributed provisioning" component=csi-driver
@ggriffiths-k8s-1-node0 portworx[14553]: time="2022-06-08T23:39:07Z" level=info msg="CSI driver gRPC connection to 192.168.121.248 is has expired after 3 minutes of no usage. Closing this connection" component=csi-driver
@ggriffiths-k8s-1-node0 portworx[14553]: time="2022-06-08T23:39:07Z" level=info msg="CSI driver gRPC connection to 192.168.121.130 is has expired after 3 minutes of no usage. Closing this connection" component=csi-driver
@ggriffiths-k8s-1-node0 portworx[14553]: time="2022-06-08T23:39:07Z" level=info msg="CSI driver gRPC connection to 192.168.121.60 is has expired after 3 minutes of no usage. Closing this connection" component=csi-driver
@ggriffiths-k8s-1-node0 portworx[14553]: time="2022-06-08T23:39:07Z" level=info msg="CSI driver gRPC connection to 192.168.121.86 is has expired after 3 minutes of no usage. Closing this connection" component=csi-driver
@ggriffiths-k8s-1-node0 portworx[14553]: time="2022-06-08T23:39:07Z" level=info msg="Cleaned up 4 connections for CSI distributed provisioning. 0 connections remaining" component=csi-driver
```